### DISCUSSION TITLE
如何监控文件的打开与关闭？

### DISCUSSION THREAD

**Discussion by hanbingzi** [opensumi/core#2695] [Q&A] [answered]
我想监控工作空间的文件打开和关闭，用来执行相应的操作，应该怎么做哪？

**Comment by erha19** [comment]
@hanbingzi 你要监听用户的编辑器操作，可以通过插件，也可以在模块内通过监听这三个事件去获取：

- EditorDocumentModelContentChangedEvent —— 编辑器内容变化
- EditorDocumentModelCreationEvent —— 编辑器文档创建
- EditorDocumentModelRemovalEvent —— 编辑器文档移除

参考用法：https://github.com/opensumi/core/issues/1494#issuecomment-1211642656

**Reply by hanbingzi** [reply]
你好，如果我想监听哪个文件被打开，或者被点击激活（包括文件树点击激活，tab框点击激活），应该监听那个？

**Comment by erha19** [comment, marked as answer]
@hanbingzi 可以用下面这种方式：

```ts
@Autowired(WorkbenchEditorService)
private readonly editorService: WorkbenchEditorServiceImpl;

this.editorService.onActiveResourceChange((resource) => {
...
})
```

额外的方法可以自己看一下 TS 声明信息
